### DISCUSSION TITLE
What does the end product do?

### DISCUSSION THREAD

**Discussion by dlucian** [personal-vault/pvp#3] [General]
My dear friend @alintuhut made a very good point when discussing the PVP project, and that is... what do you need it to do?

I think we need to answer this in the most articulated way possible, and then backtrack to what's the technical solution going to look like. "Start with the end in mind" sais [Stephen Covey](https://legacybizleaders.com/wp-content/uploads/2014/07/7-Habits-Start-with-the-end-in-mind.docx).

My very-high up, top-level, requirements are:

1) easily save files and folders

2) easily find information

Everything else falls into one of these two. 

For point 1, I want it to be very easy to save any kind of files and folders, such as (but not limited to): photos, email archives, documents, videos, software, chat archives, PDFs, projects and exports from various services like Strava, Trello, GitHub etc.

The goal is to easily save *all* my digital life, so that I have it whatever happens to the services I'm using. Also, if it's not very easy to save it, I'll probably postpone doing it until I completely give up, so it *must* be easy to save/import/copy. 

Also, data from sevices that have APIs might become files here. 

For point 2, I want it to be very easy to find information, meaning for example:
- my photos from Lisbon in June 2022
- what did I promise to John?
- what 180 BPM music do I have?
- what references do I have on the "cognitive dissonance" topic?
- what happened on this day, 1 year ago? how about 2,3,4,or 10 years ago?
- do I know Mark? when did we meet?
- what photos do I have in Sighisoara, Romania?
- what workout music do I have?
- have I every been to Lyon, France?
- when was the last time I celebrated something?
- what resources do I have about meditation?

These bits and pieces of info should be provided in multiple ways:
- simple interface with a search query like the ones above (probably web?) listing all the items found relevant to my query.
- chat bot (such as Telegram) that I can ask a question and get a short answer (+ maybe a link to said web interface?)
- API? maybe I have an integration with Siri or Alexa or something.

Very curious to hear what you think? What would PVP do for *you*?

**Comment by dlucian** [comment]
One more idea, if I have my screen recorder that gets indexed via OCR, whenever I log in or have some sort of a start-page that **answers the question "What are the files that I need to resume doing what I was doing?"**. Or even ability to ask Raycast/Alfred to "Resume" and for it to propose what I need to resume. 

Another thing is to be able to ask "What's on my todo list for today?"

**Comment by donangel** [comment]
I liked *a lot* the proposed "search strings" above. It'd be great to have this functionality.

I will also appreciate more "form search" mode, but being able to ask directly in a human-readable form will be a very nice functionality.

**Reply by dlucian** [reply]
I looks like full-text search using natural language is quite easy to achieve with [embeddings](https://supabase.com/blog/openai-embeddings-postgres-vector), so looking towards the future, this should be more standard than learning some weird syntax or yet-another-SQL. 

And if the search engine knows SQL only, there should be a software that translates "pictures with cats from last year" into "has: cats start_date:2022-01-01 end_date:2022-12-31" so we don't have to. 

Anyway, the ideal storage place is the one where you can throw anything in and easily find whatever you're looking for, even years later.
